src/main/java/com/google/sps/servlets/SearchMealServlet.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+
+@WebServlet("/search-meal")
```

Comment:
Please move this endpoint from "/search-meal" to another place as per instructions below.

This endpoint is not very RESTful, and REST is a good convention to follow for this project - it fits the usecase for every functionality you need your servlets for.
According to a [Stackoverflow comment](https://softwareengineering.stackexchange.com/questions/233164/how-do-searches-fit-into-a-restful-interface) that I found after comparing multiple approaches proposed, the best solution is to put search at "/meal" endpoint (that represents where you can get resources called "meal" from your server), and this endpoint will be **shared** with the one that just retrieves data for a chosen meal (for example - for when you open the meal's main page), that **Olga put in her code** in the pull request #2 . They both would also be a GET operation in HTTP terms.
The difference will be that for the search you'd need to pass the search params as GET request params, i.e. the URL will look like this: `mealassistant.com/meal?query=tomato%20lasagne`. Meanwhile the request for a single entry might have something like entry ID right in the URL (i.e. `/meal/12345`). You'll need some code that can tell these situations apart, and you'll need to share that place with Olga.

Reply:
So as I understand, we create only one servlet with REST methods?

Reply:
I'm not sure if you mean that, but I did similar here: 
https://github.com/sandatsian/Database-Problems/blob/master/BDProblems/Controllers/ProblemsController.cs 
on C#. Here I have a list of methods for class `Problem` and in each method has `Id` in params, so Url looks like `Problems/Edit/<Id>` when I click Edit button.

Reply:
I want it to look like this:
* `GET meal/<Id>` gets info of the meal,
* `GET meal/?query=tomato` gets search result for meals where the keyword was "tomato"
With real REST API, it could be extended to:
* `GET meal/` gets a list of meals (i.e. simplification of the search functionality),
* `POST meal/` inserts a new meal (its ID and other details are in POST request body).

Doing `Problems/Edit/<Id>` is good as long as it's URL for a nice human interface. While you can have URLs like that for managing user-accessible sites, right now we're talking about URLs for data management - for non-human exchange between frontend app (written in Javascript) and the backend app (written in Java). For that we're aiming for some standarized approach (REST) to have some simple URLs (and use various HTTP "verbs": `GET, PUT, PATCH, DELETE`) to define this software-to-software interface.
So you can have an (admin) user accessing the `/meals-page/edit/<Id>` site, which is just an html+Javascript without any servlet behind it, and then, once all edits are made, the Javascript calls `PATCH /meal/<Id>` to actually write the change.